src/request/params/query_param.rs

```diff
@@ -0,0 +1,75 @@
+use std::borrow::Cow;
+
+use super::{query_parser::QueryParser, DecodeParam, Param, PercentDecoder};
+
+pub struct QueryParam<'a, 'b> {
```

Comment:
I'm considering making `QueryParam` generic of `DecodeParm` similar to `Param` and then providing an `encoded` or `decode` method that moves self into a `QueryParam<PercentDecode>`. This would give users the choice which could be beneficial if percent-encoded query parameters are not expected. It would also allow us to add a `PercentDecodeLossy` decoder to give users a Via the choice.